### PR TITLE
Fixed spells not transferring prerequisites when dragged onto charsheet from library

### PR DESCRIPTION
**Pull request by rinickolous**

### CODE REVIEW

**Comment by ronnke** [comment]
When adding a spell to the character sheet I get the following script error.

Script Error: [string "scripts/manager_actor2.lua"]:491: attempt to index local 'skillData' (a nil value)

debugging the skillData structure I get the following.

nil
{ s'adjustedLevel' = #5, s'relativeLevel' = s'IQ-6', s'abbreviatedSkillType' = s'IQ/H' }
{ s'adjustedLevel' = #9, s'relativeLevel' = s'IQ-2', s'abbreviatedSkillType' = s'IQ/H' }

Adding a RETURN to the IF statement at line 487 of manager_actor2.lua resolves the script error, but I'm not sure that is the operation you intended when skillData is nil. 

if skillData == nil then
	-- the type or points could be invalid.
	return;
end

The spell I was trying to add was as follows.

image

**Comment by ronnke** [comment]
Ignore my previous comment, this issue is not related to your changes.

File: campaign/scripts/charsheet/char_spell_detail.lua [hunk 1]
```diff
@@ -17,7 +17,6 @@ function update()
 	self["resist"].update(bReadOnly);
 	self["text"].update(bReadOnly);
 	self["duration"].update(bReadOnly);
-	self["spellcost"].update(bReadOnly);
```
Comment (ronnke) [inline campaign/scripts/charsheet/char_spell_detail.lua:20]: Is there a particular reason you want to separate the the spell cost out from the cost/maintain field?

Comment (rinickolous) [reply campaign/scripts/charsheet/char_spell_detail.lua:20]: Oops, this must've been an oversight. I don't want to separate the fields. Will fix in a sec.

Comment (rinickolous) [reply campaign/scripts/charsheet/char_spell_detail.lua:20]: Hold on, perhaps one of us misunderstood.

The following is my understanding:
In the Ability library, the property for spell cost is "spellcost" with the label "Cost".
In the character sheet, the property for spell cost is "costmaintain" with the label "Cost/Maintain".

My assumption is that these fields are equivalent, just with different names and labels, and that the field deals with both base cost and maintain cost.

Am I understanding correctly?

Comment (ronnke) [reply campaign/scripts/charsheet/char_spell_detail.lua:20]: Yes, correct. They are equivalent, just with just with different labels.

Comment (rinickolous) [reply campaign/scripts/charsheet/char_spell_detail.lua:20]: Then I'm not separating the fields at all, I'm joining them back together. I think a previous commit separated them.

Comment (ronnke) [reply campaign/scripts/charsheet/char_spell_detail.lua:20]: Indeed you are. My bad. 👍

**Pull request merged**